Fix for ParamNames

anihex opened the pull request:
I noticed that the function ParamNames gets all param names, but doesn't remove the : at the beginning of a parameter. If one tries to use the names returned by ParamNames, there will be additional :
I added a simple TrimPrefix to solve this.

husobee commented:
Thanks for this, but honestly, I feel like this might break some that are using ParamNames currently, and it was unfortunate that we didn't TrimPrefix originally.  Would you be opposed to making a new function definition, maybe "TrimmedParamNames" or something to that extent, that calls ParamNames and performs the string trimming on the result instead?  Just worried about people who might already be using that function.

anihex commented:
I'm fine with whatever works for you. Want me to make a new PR with a new function? I mean ... It was so trivial even I was able to solve this. :smile:

anihex commented:
I cloned the new function and renamed it. Afterwards I changed the original function back to it's original state.
I also changed the function description so one can distinguish them from one another.

I hope that helps a bit.

husobee commented:
Awesome, Looks good, just need a test or two so that coveralls will not complain about coverage lacking.

anihex commented:
Added a new file which contains a hopefully useful test for the "new" function

husobee commented:
LGTM, merging.
